setup.py

```diff
@@ -28,6 +28,12 @@
 import sys
 import os
 
+# just import these always and fail early if not present
+import distutils
+from setuptools import setup
+
+IS_PYPY = '__pypy__' in sys.builtin_module_names
```

Comment:
👍

An aside... I still wonder if some distros would still prefer not to have to depend on setuptools. I guess it's so widespread now they don't care.